File: app/src/main/java/com/darakeon/kolinit/HappyActivity.java
package com.darakeon.kolinit;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class HappyActivity extends AppCompatActivity {

    public static Boolean ImHappy = true;

    public String MyNameIs = null;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.happy);

        Button happy_button = findViewById(R.id.happy_button);

        happy_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try
                {
                    showText("Hey, listen!");
                } catch (Exception e)
                {
                    e.printStackTrace();
                }
            }
        });
    }

    private void showText(String text) throws Exception
    {
        if (text == null)
            throw new Exception();

        Toast.makeText(
            this,
            text,
            Toast.LENGTH_SHORT
        ).show();
    }
}

File: app/src/main/java/com/darakeon/kolinit/HappyActivity.kt
package com.darakeon.kolinit

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import kotlinx.android.synthetic.main.happy.happy_button

class HappyActivity : AppCompatActivity() {

	var MyNameIs: String? = null

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.happy)

		happy_button.setOnClickListener {
			showText("Hey, listen!")
		}
	}

	private fun showText(text: String) {
		Toast.makeText(
			this,
			text,
			Toast.LENGTH_SHORT
		).show()
	}

	companion object {
		var ImHappy: Boolean? = true
	}
}
